haskell-src/Concordium/Types/SeedState.hs

```diff
@@ -53,36 +51,103 @@ withIsSeedStateVersionFor :: SProtocolVersion pv -> (IsSeedStateVersion (SeedSta
 withIsSeedStateVersionFor spv = withSingI (sSeedStateVersionFor spv)
 
 -- |State for computing the leadership election nonce.
-data SeedState (ssv :: SeedStateVersion) = SeedState
-    { -- |Number of slots in an epoch. This is derived from genesis
-      -- data and must not change.
-      epochLength :: !(Conditionally (SupportsEpochLength ssv) EpochLength),
-      -- |Current epoch
-      epoch :: !Epoch,
-      -- |Current leadership election nonce
-      currentLeadershipElectionNonce :: !LeadershipElectionNonce,
-      -- |The leadership election nonce updated with the block nonces
-      -- of blocks in the first 2/3 of the current epoch.
-      updatedNonce :: !Hash
-    }
-    deriving (Eq, Show)
+--
+-- (Implementation note: the constructors have equational constraints so that they have the same
+-- return time. This allows them them to have common field names (e.g. 'epoch'), which is not
```

Comment:
Typo "them them".

I don't understand what is meant here by "return time".

Reply:
Should be "return type".